src/v/cluster/types.h

```diff
@@ -1222,4 +1222,19 @@ struct adl<cluster::topic_properties> {
     void to(iobuf&, cluster::topic_properties&&);
     cluster::topic_properties from(iobuf_parser&);
 };
+
+template<typename T>
+struct adl<cluster::property_update<T>> {
```

Comment:
nit: Maybe these changes that just add adl specializations could go in a separate PR and just be pushed to dev quickly